src/input/element-input.js

```diff
@@ -607,32 +616,39 @@ pc.extend(pc, function () {
             if (button) {
                 var hitPadding = element.entity.button.hitPadding || ZERO_VEC4;
 
-                var paddingLeft = hitPadding.data[0] * scaleX;
-                var paddingBottom = hitPadding.data[1] * scaleY;
-                var paddingRight = hitPadding.data[2] * scaleX;
-                var paddingTop = hitPadding.data[3] * scaleY;
+                _paddingTop.copy(element.entity.up);
+                _paddingBottom.copy(_paddingTop).scale(-1);
+                _paddingRight.copy(element.entity.right);
+                _paddingLeft.copy(_paddingRight).scale(-1);
 
-                var cornerBottomLeft = hitCorners[0].clone();
-                var cornerBottomRight = hitCorners[1].clone();
-                var cornerTopRight = hitCorners[2].clone();
-                var cornerTopLeft = hitCorners[3].clone();
+                _paddingTop.scale(hitPadding.data[3] * scaleY);
+                _paddingBottom.scale(hitPadding.data[1] * scaleY);
+                _paddingRight.scale(hitPadding.data[2] * scaleX);
+                _paddingLeft.scale(hitPadding.data[0] * scaleX);
 
-                cornerBottomLeft.x -= paddingLeft;
-                cornerBottomLeft.y -= paddingBottom;
+                _cornerBottomLeft.copy(hitCorners[0]).add(_paddingBottom).add(_paddingLeft);
+                _cornerBottomRight.copy(hitCorners[1]).add(_paddingBottom).add(_paddingRight);
+                _cornerTopRight.copy(hitCorners[2]).add(_paddingTop).add(_paddingRight);
+                _cornerTopLeft.copy(hitCorners[3]).add(_paddingTop).add(_paddingLeft);
 
-                cornerBottomRight.x += paddingRight;
-                cornerBottomRight.y -= paddingBottom;
+                hitCorners = [_cornerBottomLeft, _cornerBottomRight, _cornerTopRight, _cornerTopLeft];
+            }
 
-                cornerTopRight.x += paddingRight;
-                cornerTopRight.y += paddingTop;
+            return hitCorners;
+        },
 
-                cornerTopLeft.x -= paddingLeft;
-                cornerTopLeft.y += paddingTop;
+        _calculateScaleToScreen: function(element) {
+            var current = element.entity;
+            var screenScale = element.screen.screen.scale;
```

Comment:
We seem to have lost the check on whether the element has a screen? is it possible for this to run on an element with no screen?